pkg/apis/edge/v1alpha1/types_location.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// LocationLabelsStringAnnotationKey is the label key for the label holding a string
+	// representation of the location labels in order to use them in a table column in the CLI.
+	LocationLabelsStringAnnotationKey = "scheduling.kcp.io/labels"
+
+	// PlacementAnnotationKey is the label key for the label holding a PlacementAnnotation struct.
+	PlacementAnnotationKey = "scheduling.kcp.io/placement"
+)
+
+// Location represents a set of instances of a scheduling resource type acting a target
+// of scheduling.
+//
+// The location is chosen by the user (in the future) through a Placement object, while
+// the instance is chosen by the scheduler depending on considerations like load
+// or available resources, or further node selectors specified by the user.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
```

Comment:
I am not sure we want to retain the "kcp" category.

Reply:
what do you recommend?

Reply:
For the other kinds of KubeStellar objects, we have not put them in any category.  I think that it would make sense to follow suit here.  We might choose later to put them all in some category, that would be done in its own PR.